app/models/goal.py

```diff
@@ -2,4 +2,14 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    def create_goal_dict(self):
```

Comment:
Style: Since this is a method on the `Goal` class having `goal` in the name is redundant:

```suggestion
    def create_dict(self):
```